### PR TITLE
Timezone support

### LINKED ISSUE #21: Add support for timezones
Add support for hammertime to be aware of applications running in different timezones so it can shutdown and start resources according to the timezone they need to run in.

### PR DESCRIPTION
**Pull request by ramesius**
Fixes #21

### CODE REVIEW

**Comment by ramesius** [comment]
Merge after #17

File: events.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const operatingTimezones = require('./operatingTimezones');
+const isEnabled = require('./isEnabled').isEnabled;
+
+const UTC_START_HOUR = 9;
```
Comment (ramesius) [inline events.js]: These constants now control the start and stop hour of hammertime

File: operatingTimezones.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const timezones = process.env.HAMMERTIME_OPERATING_TIMEZONES || '[10]';
+
+const parsed = JSON.parse(timezones);
```
Comment (ramesius) [inline operatingTimezones.js]: Timezones should be express as a javascript array in the environment variable.

File: src/tags/operatingTimezone.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * specified where in that the current UTC date is between the specified dates.
+ * @param {Array<{Key: string, Value: string}>} tags
+ * @returns {number}
+ */
+module.exports = tags => {
+  const timezoneTags = getHammertimeTags(tags).filter(tag => caseInvariantStringEquals(tag.Key, 'hammertime:operatingTimezone'));
+
+  if (timezoneTags.length === 0 || timezoneTags.length > 1) {
+    console.log('No timezone tag found or multiple timezone tags found. Defaulting to +10');
+    return 10;
```
Comment (ramesius) [inline src/tags/operatingTimezone.js]: Default to AEDT (+10) for resources that do not have a `hammertime:operatingTimezone` tag

File: events.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      currentOperatingTimezone: timezone,
+    },
+  }));
+  return {
+    schedule: startCrons, // CRONS for starting
+  };
+}
+
+module.exports.stop = stop;
+module.exports.start = start;
```
Comment (ramesius) [inline events.js]: These functions generate the schedule events that end up being put into the serverless.yml on deployment. It works similar to how the dryRun and isEnabled environment variable javascript files work.

File: src/tags/operatingTimezone.js (hunk 3 above)
Comment (krutisfood) [reply src/tags/operatingTimezone.js]: Config file would be ideal but the console log is nice and clear, the filename seems obvious enough, nice!

File: test/asgs/listASGsToStart.test.js [hunk 5]
```diff
@@ -9,7 +9,7 @@ describe('listASGsToStart()', () => {
   it('returns list of asgs spun down by hammertime', () => {
     AWS.mock('AutoScaling', 'describeAutoScalingGroups', startOnePageResponse);
 
-    return listASGsToStart()
+    return listASGsToStart(10)
```
Comment (krutisfood) [inline test/asgs/listASGsToStart.test.js]: This would be more obvious as a variable called "defaultTimeZone", thoughts?

Comment (ramesius) [reply test/asgs/listASGsToStart.test.js]: Indeed, will sort that out.

**Comment by krutisfood** [comment]
needs a version bump pls.

**Comment by krutisfood** [comment]
Can we change this PR to master now?
Also needs some updates to the README so we know how it works :)
Early testing is looking good though, we've got multiple cloudwatch events with two timezones configured 👍

**Comment by ramesius** [comment]
Yeah I think a nice write up in the readme will definetly be worth it. Also after some thinking I would like to see if I can include configuration for timezones using the format 'Australia/Sydney' so DST can be taken into account when deciding to start/stop an asset.

**Comment by ramesius** [comment]
`m().tz('Australia/Sydney').utcOffset()` I think this will suffice getting the timezone offeset, will implement this soon.

**Comment by ramesius** [comment]
Have fixed up a few rough edges in the code. Switched out moment for luxon which provided a nicer API for dealing with the date and time stuff. Added the ability to configure the start and stop time of hammertime. Have fixed up some things in the readme also.

**Comment by krutisfood** [comment]
For some reason hasn't picked up any of the ASGs, though happily spun down our instances
```11:01:24 ... No ASGs to spin down, moving on...```

File: .travis.yml [hunk 6]
```diff
@@ -1,5 +1,8 @@
 language: node_js
 node_js:
-  - "node"
+  - "8"
```
Comment (krutisfood) [inline .travis.yml]: Bit late to the party, but should this reflect the version in serverless.yml? or are we just anticipating the next PR :)

File: serverless.yml [hunk 7]
```diff
@@ -33,20 +33,13 @@ provider:
       Resource: 'arn:aws:logs:*:*:*'
   environment:
     hammerTimeDryRun: '${file(./isDryRun.js):isDryRun}'
-  custom:
-    enabled: '${file(./isEnabled.js):isEnabled}'
+    defaultOperatingTimezone: '${file(./operatingTimezones.js):defaultOperatingTimezone}'
 functions:
   stop-hammertime:
     handler: hammertime.stop
     timeout: 180
```
Comment (krutisfood) [inline serverless.yml]: We need a longer timeout these days, could be subsequent PR though to make it a parameter :)

**Comment by krutisfood** [comment]
Spun down 316 ASGs and half dozen instances, nice! We'll see how well it boots everything up in the morning & must be time to put this PR to bed!

**Pull request merged**